src/app/applications/datatarget/opendatadk/opendatadk-edit/opendatadk-edit.component.html

```diff
@@ -1,86 +1,384 @@
+                  <mat-select
+                    multiple
+                    name="devices"
+                    [compareWith]="payloadDevicesDropdownCompare"
+                    [(value)]="element.iotDeviceIds">
+                    <mat-option disabled="disabled" class="filter-option">
+                      <button
+                        mat-raised-button
+                        class="mat-primary fill text-sm"
+                        (click)="selectAllDevices(i)">
+                          {{ 'QUESTION.DATATARGET.SELECTALLDEVICES' | translate }}
+                      </button>
+                      <button
+                        mat-raised-button
+                        class="mat-primary fill text-sm"
+                        (click)="deSelectAllDevices(i)"
+                      >
+                        {{
+                          'QUESTION.DATATARGET.DESELECTALLDEVICES' | translate
+                        }}
+                      </button>
+                    </mat-option>
+                    <mat-option
+                      *ngFor="let device of devices"
+                      [value]="device.id"
+                      >{{ device.name }}</mat-option
+                    >
+                  </mat-select>
+                </mat-form-field>
+              </div>
+            </td>
+            <td>
+              <div class="row">
+                <mat-form-field appearance="fill">
+                  <mat-label>{{
+                    'QUESTION.DATATARGET.SELECT-PAYLOADDECODER' | translate
+                  }}</mat-label>
+                  <mat-select
+                    matNativeControl
+                    name="payloadDecoderId"
+                    [(value)]="element.payloadDecoderId"
+                  >
+                    <mat-option [value]="0">
+                      {{
+                        'QUESTION.DATATARGET.NO-PAYLOAD-DECODER-SELECTED'
+                          | translate
+                      }}
+                    </mat-option>
+                    <mat-option
+                      *ngFor="let payloadDecoder of payloadDecoders"
+                      [value]="payloadDecoder.id"
+                    >
+                      {{ payloadDecoder.name }}
+                    </mat-option>
+                  </mat-select>
+                </mat-form-field>
+              </div>
+            </td>
+            <td>
+              <a (click)="openDeleteDialog(i)">
+                <div class="text-center m-2">
+                  <fa-icon [icon]="faTimesCircle"></fa-icon>
+                  <p>{{ 'DATATARGET.DELETE' | translate }}</p>
+                </div>
+              </a>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </ng-container>
+  </div>
+
+  <!-- Save and back-buttons -->
+  <div class="form-group mt-5">
+    <button (click)="routeToDatatargets()" class="btn btn-light" type="button">
```

Comment:
This does not actually route to data targets, but to IoT-devices

Reply:
Fair